pkg/proxy/kafka.go

```diff
@@ -258,24 +351,53 @@ func handleConnection(pair *connectionPair, c *proxyConnection, handler kafkaMes
 	}
 }
 
+func handleResponse(pair *connectionPair, c *proxyConnection,
+	record *kafkaLogRecord, handler kafkaRespMessageHander) {
+	for {
+		rsp, err := kafka.ReadResponse(c.conn)
+		if err != nil {
+			if err == io.EOF || err == io.ErrUnexpectedEOF {
+				c.Close()
+				return
+			}
+
+			if record != nil {
+				record.log(accesslog.TypeResponse, accesslog.VerdictError,
+					kafka.ErrInvalidMessage,
+					fmt.Sprintf("Unable to parse Kafka response: %s", err))
+			}
+
+			log.WithError(err).Error("Unable to parse Kafka response")
+			continue
+		} else {
+			handler(pair, rsp)
+		}
+	}
+}
+
 func (k *kafkaRedirect) handleRequestConnection(pair *connectionPair) {
 	log.WithFields(log.Fields{
 		"from": pair.rx,
 		"to":   pair.tx,
 	}).Debug("Proxying request Kafka connection")
 
-	handleConnection(pair, pair.rx, k.handleRequest)
+	handleRequest(pair, pair.rx, nil, k.handleRequest)
 }
 
-func (k *kafkaRedirect) handleResponseConnection(pair *connectionPair) {
+func (k *kafkaRedirect) handleResponseConnection(pair *connectionPair,
+	record *kafkaLogRecord) {
 	log.WithFields(log.Fields{
 		"from": pair.tx,
 		"to":   pair.rx,
 	}).Debug("Proxying response Kafka connection")
 
-	handleConnection(pair, pair.tx, func(pair *connectionPair, req *kafka.RequestMessage) {
-		pair.rx.Enqueue(req.GetRaw())
+	handleResponse(pair, pair.tx, record, func(pair *connectionPair,
```

Comment:
It might be more clear if we split the line before the func

Reply:
fixed